client/src/components/Filter.jsx

```diff
@@ -35,19 +35,11 @@ const Filter = ({ filterLabel }) => {
     2 - some or all boxes are initially checked, and changes are made
   */
   const [buttonStatus, setButtonStatus] = useState(() => {
-    Object.keys(checked).forEach((key) => {
-      if (checked[key].checked) return 1;
-    });
-    return 0;
+    return Object.values(checked).find((item) => item.checked) ? 1 : 0;
   });
 
   const handleClick = (event) => {
-    let hasAtLeastOneChecked = false;
-    Object.keys(checked).forEach((key) => {
-      if (checked[key].checked) hasAtLeastOneChecked = true;
-    });
-
-    if (hasAtLeastOneChecked) {
+    if (Object.values(checked).find((item) => item.checked)) {
       setButtonStatus(1);
```

Comment:
SetButtonStatus takes a callback that can simplify this code!

setButtonStatus(() => Object.keys...)